js/id/ui/preset/wikipedia.js

```diff
@@ -105,9 +107,23 @@ iD.ui.preset.wikipedia = function(field, context) {
             title.value(value);
         }
 
-        var t = {};
-        t[field.key] = value ? language()[2] + ':' + value : undefined;
-        event.change(t);
+        syncTags[field.key] = value ? language()[2] + ':' + value : undefined;
+        event.change(syncTags);
+        
+        if (value && language()[2]) {
+            wikidata.itemsByTitle(language()[2], value, function (title, entities) {
+                var qids = entities && Object.keys(entities),
+                    qid = qids && _.find(qids, function (id) {
+                    return id.match(/^Q\d+$/);
+                }),
+                    selectedIDs = context.selectedIDs && context.selectedIDs();
+                if (selectedIDs && selectedIDs.length === 1 && selectedIDs[0] === entity.id) {
+                    event.change({wikidata: qid});
+                } else {
+                    context.entity(entity.id).tags.wikidata = qid;
```

Comment:
Cool, @paulmach yeah your code is the right way to do things, and is safer because you use `iD.ui.Loading` to stop the user from editing while waiting for the callback.  

It would be even *more* bulletproof if you do
`iD.ui.Loading(context).blocking(true).message('Sliding')`  
because without the `.blocking(true)` the user can just click and make the dialog go away.  I was able to slide a really long twisty way, dismiss the dialog, and delete the way, then the callback happened and threw an "entity xxx not found" error.  Kind of a tricky sequence, but a user haphazardly clicking around the radial menu could do it.